packages/node/src/indexer/fetch.service.ts

```diff
@@ -113,14 +136,42 @@ export function buildDictionaryQueryEntries(
       if (!handler.filter) return [];
 
       switch (handler.kind) {
-        case StellarHandlerKind.Event: {
-          const filter = handler.filter as StellarEventFilter;
+        case StellarHandlerKind.Transaction: {
+          const filter = handler.filter as StellarTransactionFilter;
+          if (ds.groupedOptions) {
+            queryEntries.push(
+              transactionFilterToQueryEntry(filter, ds.groupedOptions),
+            );
+          } else if (filter.account) {
+            queryEntries.push(
+              transactionFilterToQueryEntry(filter, ds.options),
+            );
+          } else {
+            return [];
+          }
+          break;
+        }
+        case StellarHandlerKind.Operation: {
+          const filter = handler.filter as StellarOperationFilter;
+          if (ds.groupedOptions) {
+            queryEntries.push(
+              operationFilterToQueryEntry(filter, ds.groupedOptions),
+            );
+          } else if (filter.source_account || filter.type) {
+            queryEntries.push(operationFilterToQueryEntry(filter, ds.options));
+          } else {
+            return [];
+          }
+          break;
+        }
+        case StellarHandlerKind.Effects: {
```

Comment:
There doesn't seem to be a case for Events.

Same for block handlers. It should behave same as with other chains